flink-runtime/src/main/java/org/apache/flink/runtime/state/SharedStateRegistryImpl.java

```diff
@@ -192,6 +208,8 @@ public void registerAll(
     @Override
     public void registerAllAfterRestored(CompletedCheckpoint checkpoint, RestoreMode mode) {
```

Comment:
This method is called by `CompletedCheckpoint#registerSharedStatesAfterRestored()`, and `registerSharedStatesAfterRestored()` is called by `CheckpointCoordinator#restoreSavepoint()` and `DEFAULT_FACTORY`.
 I'm not sure if the call in `DEFAULT_FACTORY` has other effects. If a job failover and restarts, does `highestRestoredCheckpointID` update?

Reply:
Yes, that would not violate correctness, but that would delay the deletion of more checkpoints(the completed checkpoints which are stored in ZK) folders.